doc/guides/west/workspaces.rst

```diff
@@ -0,0 +1,309 @@
+.. _west-workspaces:
+
+Workspaces
+##########
+
+This page describes the *west workspace* concept introduced in
+:ref:`west-basics` in more detail.
+
+.. _west-manifest-rev:
+
+The ``manifest-rev`` branch
+***************************
+
+West creates and controls a Git branch named ``manifest-rev`` in each
+project. This branch points to the revision that the manifest file
+specified for the project at the time :ref:`west-update` was last run.
+Other workspace management commands may use ``manifest-rev`` as a reference
+point for the upstream revision as of this latest update. Among other
+purposes, the ``manifest-rev`` branch allows the manifest file to use SHAs
+as project revisions.
+
+Although ``manifest-rev`` is a normal Git branch, west will recreate and/or
+reset it on the next update. For this reason, it is **dangerous**
+to check it out or otherwise modify it yourself. For instance, any commits
+you manually add to this branch may be lost the next time you run ``west
+update``. Instead, check out a local branch with another name, and either
+rebase it on top of a new ``manifest-rev``, or merge ``manifest-rev`` into
+it.
```

Comment:
... Some people find useful to make their branches track `manifest-rev`:
```
git branch --set-upstream-to=manifest-rev
git status
Your branch and 'manifest-rev' have diverged,
and have 1 and 2 different commits each, respectively.
```
```

Reply:
> ... Some people find useful to make their branches track `manifest-rev`:

Maybe we can make a 'tips and tricks' page sometime, but I'm not going to add this recommendation in this PR.